src/test/scala/org/broadinstitute/dsde/firecloud/service/LibraryApiServiceSpec.scala

```diff
@@ -0,0 +1,103 @@
+package org.broadinstitute.dsde.firecloud.service
+
+import java.net.URL
+
+import org.broadinstitute.dsde.firecloud.Application
+import org.broadinstitute.dsde.firecloud.dataaccess._
+import org.broadinstitute.dsde.firecloud.mock.MockUtils
+import org.broadinstitute.dsde.firecloud.mock.MockUtils._
+import org.broadinstitute.dsde.firecloud.model.UserInfo
+import org.broadinstitute.dsde.firecloud.webservice.LibraryApiService
+import org.mockserver.integration.ClientAndServer
+import org.mockserver.integration.ClientAndServer._
+import org.mockserver.model.HttpRequest._
+import org.parboiled.common.FileUtils
+
+import scala.util.Try
+import spray.http.HttpMethods
+import spray.http.StatusCodes._
+import spray.json._
+import spray.util._
+import org.everit.json.schema.Schema
+import org.everit.json.schema.loader.SchemaLoader
+import org.json.JSONObject
+import org.json.JSONTokener
+
+class LibraryApiServiceSpec extends ServiceSpec with LibraryApiService {
+
+  def actorRefFactory = system
+  var workspaceServer: ClientAndServer = _
+
+  lazy val isCuratorPath = "/api/library/user/role/curator"
+
+  val rawlsDAO:RawlsDAO = new MockRawlsDAO
+  val searchDAO:SearchDAO = new ElasticSearchDAO
+  val app:Application = new Application(rawlsDAO, searchDAO)
+  val libraryServiceConstructor: (UserInfo) => LibraryService = LibraryService.constructor(app)
+
+  override def beforeAll(): Unit = {
+
+    workspaceServer = startClientAndServer(workspaceServerPort)
+    workspaceServer
+      .when(request.withMethod("GET").withPath(isCuratorPath))
+      .respond(
+        org.mockserver.model.HttpResponse.response()
+          .withHeaders(MockUtils.header).withStatusCode(OK.intValue)
+      )
+  }
+
+  override def afterAll(): Unit = {
+    workspaceServer.stop()
+  }
+
+  "LibraryService" - {
+
+    /* Handle passthrough handlers here */
+
+    "when calling the isCurator endpoint" - {
+      "PUT, POST, DELETE on /api/library/user/role/curator" - {
+        "should receive a MethodNotAllowed" in {
+          List(HttpMethods.PUT, HttpMethods.POST, HttpMethods.DELETE) map {
+            method =>
+              new RequestBuilder(method)(isCuratorPath) ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
+                System.out.println(response)
+                status should equal(MethodNotAllowed)
+              }
+          }
+        }
+      }
+    }
+
+    "in its schema definition" - {
+      "has valid JSON" in {
+        val classLoader = actorSystem(actorRefFactory).dynamicAccess.classLoader
```

Comment:
Do we need a whole classloader? In other code, we use `getClass.getResource(...)` 
Definitely not a big deal in a test.